other_solutions/dakshayahuja.py

```diff
@@ -0,0 +1,141 @@
+from tkinter import *
+import tkinter.messagebox
+import random
+
+root = Tk()
+
+label1 = Label(root, text="Welcome to T.T.T", bg='blue', fg='white')
+label1.pack()
+
+tkinter.messagebox._show('Welcome Message', 'Welcome to T.T.T')
+
+
+def display_board(board):  # this function is for displaying grids and marks for user convenience
+    print('   |   |   ')
+    print(' ' + board[7] + ' | ' + board[8] + ' | ' + board[9])
+    print('   |   |   ')
+    print('-----------')
+    print('   |   |   ')
+    print(' ' + board[4] + ' | ' + board[5] + ' | ' + board[6])
+    print('   |   |   ')
+    print('-----------')
+    print('   |   |   ')
+    print(' ' + board[1] + ' | ' + board[2] + ' | ' + board[3])
+    print('   |   |   ')
+
+
+def player_input():  # this function is for asking user to make a choice ie., like toss to start game
+    marker = ''
+    while not (marker == 'X' or marker == 'O'):
+        marker = input('Player 1: Do you want to be X or O :').upper()
+
+    if marker == 'X':
+        return ('X', 'O')
+    else:
+        return ('O', 'X')
+
+
+def place_marker(board, marker, position):
+    board[position] = marker
+
+
+def win_check(board, mark):  # this function is for checking whether the player is win or nor
+    return ((board[7] == mark and board[8] == mark and board[9] == mark) or  # if all the marks in top row are equal
+            (board[4] == mark and board[5] == mark and board[6] == mark) or  # if all the marks in middle row are equal
+            (board[1] == mark and board[2] == mark and board[3] == mark) or  # if all the marks in bottom row are equal
+            (board[7] == mark and board[4] == mark and board[1] == mark) or  # if all the marks in first column are equal
+            (board[8] == mark and board[5] == mark and board[2] == mark) or  # if all the marks in second column are equal
+            (board[9] == mark and board[6] == mark and board[3] == mark) or  # if all the marks in third column are equal
+            (board[7] == mark and board[5] == mark and board[3] == mark) or  # if all the marks in left to right daigonal
+            (board[9] == mark and board[5] == mark and board[1] == mark))   # if all the marks in left to right daigonal
+
+
+def choose_first():  # this function is for selecting one random player to play first turn
+    if random.randint(0, 1) == 0:
+        return 'Player 2'
+    else:
+        return 'Player 1'
+
+
+def space_check(board, position):  # this function is used for checking whether a space is available or not in board
```

Comment:
A lot of lines exceed 80 character limits, if it's about 85 characters that's ok but if the line gets to near 90 characters, split the line.  
Also, instead of adding comment in front of the function write it on the line above it as this is a very long comment.